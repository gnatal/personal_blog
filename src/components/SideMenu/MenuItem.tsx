import React from 'react';

function MenuItem({
  linkURL,
  linkLabel,
}: {
  linkURL: string;
  linkLabel: string;
}) {
  return (
    <div className='w-screen pl-6 pb-2 pt-2 border'>
      <a
        href={linkURL}
        className='block font-bold text-md text-teal-100 hover:text-white'
      >
        {linkLabel}
      </a>
    </div>
  );
}

export default MenuItem;
