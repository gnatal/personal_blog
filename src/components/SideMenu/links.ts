const about = {
  linkURL: '/about',
  linkLabel: 'sobre',
};

const login = {
  linkURL: '/login',
  linkLabel: 'login',
};

const home = {
  linkURL: '/',
  linkLabel: 'in√≠cio',
};

const posts = {
  linkURL: '/posts',
  linkLabel: 'blog',
};

export default {
  about,
  posts,
  login,
  home,
};
