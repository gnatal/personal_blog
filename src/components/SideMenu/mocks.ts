const link1 = 'asd';

export default {
  link1,
};
