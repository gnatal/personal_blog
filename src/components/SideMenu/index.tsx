import React from 'react';
import MenuItem from './MenuItem';
import links from './links';

function SideMenu({ isOpen }: { isOpen: boolean }) {
  return (
    <div
      className={`${
        isOpen ? 'block' : 'hidden'
      } sm:hidden bg-teal-500 h-screen w-screen`}
    >
      <MenuItem linkURL={links.home.linkURL} linkLabel={links.home.linkLabel} />
      <MenuItem
        linkURL={links.about.linkURL}
        linkLabel={links.about.linkLabel}
      />
      <MenuItem
        linkURL={links.posts.linkURL}
        linkLabel={links.posts.linkLabel}
      />
      <MenuItem
        linkURL={links.login.linkURL}
        linkLabel={links.login.linkLabel}
      />
    </div>
  );
}

export default SideMenu;
