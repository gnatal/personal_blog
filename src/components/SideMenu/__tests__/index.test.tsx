import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import SideMenu from '..';

afterEach(cleanup);
test('It should show open  menu', () => {
  const component = renderer.create(<SideMenu isOpen />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('It should show close the menu', () => {
  const component = renderer.create(<SideMenu isOpen={false} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('It should open and close menu', () => {
  const { container } = render(<SideMenu isOpen={false} />);
  expect(container.getElementsByClassName('hidden')).toBeTruthy();
});

test('It should open and close menu', () => {
  const { container } = render(<SideMenu isOpen />);
  expect(container.getElementsByClassName('block')).toBeTruthy();
});
