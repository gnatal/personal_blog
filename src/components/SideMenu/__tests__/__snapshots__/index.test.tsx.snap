// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should show close the menu 1`] = `
<div
  className="hidden sm:hidden bg-teal-500 h-screen w-screen"
>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/"
    >
      início
    </a>
  </div>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/about"
    >
      sobre
    </a>
  </div>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/posts"
    >
      blog
    </a>
  </div>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/login"
    >
      login
    </a>
  </div>
</div>
`;

exports[`It should show open  menu 1`] = `
<div
  className="block sm:hidden bg-teal-500 h-screen w-screen"
>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/"
    >
      início
    </a>
  </div>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/about"
    >
      sobre
    </a>
  </div>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/posts"
    >
      blog
    </a>
  </div>
  <div
    className="w-screen pl-6 pb-2 pt-2 border"
  >
    <a
      className="block font-bold text-md text-teal-100 hover:text-white"
      href="/login"
    >
      login
    </a>
  </div>
</div>
`;
